agent.go

```diff
@@ -114,10 +114,11 @@ type process struct {
 }
 
 type container struct {
-	container libcontainer.Container
-	config    configs.Config
-	processes map[string]*process
-	pod       *pod
+	container     libcontainer.Container
+	config        configs.Config
+	processes     map[string]*process
+	pod           *pod
+	processesLock sync.RWMutex
```

Comment:
I would call this one `containerLock` instead of `processesLock` since it locks the entire container.

Reply:
Fair enough.